src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -198,5 +212,22 @@ private string GetDebuggerDisplay()
         {
             return nameof(Path) + " = " + Path;
         }
+
+        private static void ValidateFileLength(Workspace workspace, string path)
+        {
+            var fileLength = FileUtilities.GetFileLength(path);
+            if (fileLength > workspace.Options.GetOption(FileTextLoaderOptions.FileLengthThreshold))
+            {
+                // log max file length to AI
+                Logger.Log(FunctionId.FileTextLoader_FileLengthThresholdExceeded, KeyValueLogMessage.Create(m =>
+                {
+                    m["FileLength"] = fileLength;
+                    m["Ext"] = PathUtilities.GetExtension(path);
```

Comment:
Will this get associated with some other project type information? My worry is there's some customer out there with some large auto-generated file (think LINQ, EF, etc.) and we'll see here it's a C# file but not know further details. Although I'm not sure what else _could_ be logged here...